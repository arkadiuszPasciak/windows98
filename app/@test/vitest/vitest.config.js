import { defineProject, mergeConfig } from "vitest/config"
import sharedConfig from "@windows98/configs/vitest/shared"
import tsconfigPaths from "vite-tsconfig-paths"

export default mergeConfig(
	sharedConfig,
	defineProject({
		plugins: [tsconfigPaths({ loose: true })],
	})
)
