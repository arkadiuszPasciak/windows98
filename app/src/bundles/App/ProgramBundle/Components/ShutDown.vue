<template>
  <UIModal
    class="ShutDown"
    :title="t('ShutDownBundle.title')"
    :resize-window="false"
    width="330px"
    height="180px"
    :modal-state="programStore.modalShutDown"
    @close-modal="closeModal"
  >
    <div class="content">
      <mf-shutdown />
    </div>
  </UIModal>
</template>

<script setup lang="ts">
import { useProgramStore } from "@APP/src/bundles/App/ProgramBundle/Stores/Program.stores"
import UIModal from "@APP|Bundles/UIModalBundle/Components/UIModal.vue"
import { useI18n } from "vue-i18n"

const { t } = useI18n()

const programStore = useProgramStore()

const closeModal = (): void => {
	programStore.updateShutDownModal(false)
}
</script>

