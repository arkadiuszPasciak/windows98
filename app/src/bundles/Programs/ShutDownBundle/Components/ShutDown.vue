<template>
  <UIModal
    class="ShutDown"
    :title="t('ShutDownBundle.title')"
    :resize-window="false"
    width="330px"
    height="180px"
    :modal-state="programStore.modalShutDown"
    @close-modal="closeModal"
  >
    <div class="content">
      <img
        class="icon"
        width="32"
        height="32"
        src="/icons/applications/mycomputer.png"
        alt="My computer"
        data-test="shut-down-icon"
      >
      <UIText
        class="description"
        data-test="shut-down-description"
      >
        {{ t('ShutDownBundle.description') }}
      </UIText>
      <UIRadio
        class="checkbox-shut-down"
        :name="shutDownRadioInputName"
        :checked="true"
        :model-value="EShutDownRadioCheck.SHUT_DOWN"
        @update:model-value="checkedRadioBoxes = $event"
      >
        {{ t('ShutDownBundle.shut-down') }}
      </UIRadio>
      <UIRadio
        class="checkbox-restart"
        :name="shutDownRadioInputName"
        :model-value="EShutDownRadioCheck.RESTART"
        @update:model-value="checkedRadioBoxes = $event"
      >
        {{ t('ShutDownBundle.restart') }}
      </UIRadio>
      <UIButton
        class="button is-ok"
        data-test="shut-down-button-ok"
        @click="triggerEvent"
      >
        {{ t('ShutDownBundle.ok') }}
      </UIButton>
      <UIButton
        class="button is-cancel"
        data-test="shut-down-button-cancel"
        @click="closeModal"
      >
        {{ t('ShutDownBundle.cancel') }}
      </UIButton>
    </div>
  </UIModal>
</template>

<script setup lang="ts">
import { useProgramStore } from "@APP/src/bundles/App/ProgramBundle/Stores/Program.stores"
import {
	restartSystem,
	shutDownSystem,
} from "@APP/src/bundles/Programs/ShutDownBundle/Services/ShutDown.services"
import {
	EShutDownRadioCheck,
	type IShutDownRadioCheck,
	shutDownRadioInputName,
} from "@APP/src/bundles/Programs/ShutDownBundle/Supports/ShutDown.supports"
import UIButton from "@APP|Bundles/UIButtonBundle/Components/UIButton.vue"
import UIModal from "@APP|Bundles/UIModalBundle/Components/UIModal.vue"
import UIRadio from "@APP|Bundles/UIRadioBundle/Components/UIRadio.vue"
import UIText from "@APP|Bundles/UITextBundle/Components/UIText.vue"
import { ref } from "vue"
import { useI18n } from "vue-i18n"

const { t } = useI18n()

const programStore = useProgramStore()

const checkedRadioBoxes = ref<IShutDownRadioCheck>(
	EShutDownRadioCheck.SHUT_DOWN,
)

const closeModal = (): void => {
	programStore.updateShutDownModal(false)
}

const triggerEvent = (): void => {
	if (checkedRadioBoxes.value === EShutDownRadioCheck.RESTART) {
		restartSystem(window)
	} else if (checkedRadioBoxes.value === EShutDownRadioCheck.SHUT_DOWN) {
		shutDownSystem(window)
	} else {
		return
	}

	closeModal()
}
</script>

<style
  lang="scss"
  scoped
  src="@APP|Bundles/ShutDownBundle/Styles/ShutDown.styles.scss"
/>
