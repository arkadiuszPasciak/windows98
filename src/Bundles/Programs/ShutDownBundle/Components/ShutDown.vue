<template>
  <UIModal
    class="ShutDown"
    :title="t('ShutDownBundle.title')"
    :resize-window="false"
    width="330px"
    height="180px"
    :modal-state="programStore.modalShutDown"
    @close-modal="closeModal"
  >
    <div class="content">
      <img
        class="icon"
        width="32"
        height="32"
        src="/icons/applications/mycomputer.png"
        alt="My computer"
        data-test="shut-down-icon"
      />
      <UIText class="description" data-test="shut-down-description">
        {{ t('ShutDownBundle.description') }}
      </UIText>
      <UIRadio
        class="checkbox-shut-down"
        :name="shutDownRadioInputName"
        :checked="true"
        :model-value="EShutDownRadioCheck.SHUT_DOWN"
        @update:model-value="checkedRadioBoxes = $event"
      >
        {{ t('ShutDownBundle.shut-down') }}
      </UIRadio>
      <UIRadio
        class="checkbox-restart"
        :name="shutDownRadioInputName"
        :model-value="EShutDownRadioCheck.RESTART"
        @update:model-value="checkedRadioBoxes = $event"
      >
        {{ t('ShutDownBundle.restart') }}
      </UIRadio>
      <UIButton
        class="button is-ok"
        data-test="shut-down-button-ok"
        @click="triggerEvent"
      >
        {{ t('ShutDownBundle.ok') }}
      </UIButton>
      <UIButton
        class="button is-cancel"
        data-test="shut-down-button-cancel"
        @click="closeModal"
      >
        {{ t('ShutDownBundle.cancel') }}
      </UIButton>
    </div>
  </UIModal>
</template>

<script setup lang="ts">
  import { ref, Ref } from 'vue'
  import { useI18n } from 'vue-i18n'
  import UIButton from '@Bundles/UIBundle/Components/UIButton.vue'
  import UIModal from '@Bundles/UIModalBundle/Components/UIModal.vue'
  import UIRadio from '@Bundles/UIBundle/Components/UIRadio.vue'
  import UIText from '@Bundles/UITextBundle/Components/UIText.vue'
  import {
    restartSystem,
    shutDownSystem,
  } from '@Bundles/ShutDownBundle/Services/ShutDown.services'
  import { useProgramStore } from '@Bundles/ProgramBundle/Stores/Program.stores'
  import {
    EShutDownRadioCheck,
    IShutDownRadioCheck,
    shutDownRadioInputName,
  } from '@Bundles/ShutDownBundle/Supports/ShutDown.supports'

  const { t } = useI18n()

  const programStore = useProgramStore()

  const checkedRadioBoxes = ref('shut-down') as Ref<IShutDownRadioCheck>

  const closeModal = (): void => {
    programStore.updateShutDownModal(false)
  }

  const triggerEvent = (): void => {
    if (checkedRadioBoxes.value === EShutDownRadioCheck.RESTART) {
      restartSystem(window)
    } else if (checkedRadioBoxes.value === EShutDownRadioCheck.SHUT_DOWN) {
      shutDownSystem(window)
    } else {
      return
    }

    closeModal()
  }
</script>

<i18n src="@Bundles/ShutDownBundle/Locales/ShutDown.locales.json"></i18n>

<style
  lang="scss"
  scoped
  src="@Bundles/ShutDownBundle/Styles/ShutDown.styles.scss"
/>
