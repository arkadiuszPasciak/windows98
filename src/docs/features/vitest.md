# [Vitest](https://vitest.dev/)

## Description

Vitest is a modern and fast testing framework specifically designed for Vite-powered projects. Developed by the [Vite team](https://vitejs.dev/team.html), Vitest leverages Vite's architecture to provide a seamless testing experience with features like native ESM support, rapid test execution, and isolated testing environments. It offers compatibility with various front-end libraries and frameworks, such as Vue, React, and Svelte. With its straightforward configuration and integration with Vite, Vitest aims to enhance the developer experience by providing an efficient and enjoyable testing workflow for Vite-based web applications.

## Commands

[Available commands in the application](/reference/cli.html#vitest)
