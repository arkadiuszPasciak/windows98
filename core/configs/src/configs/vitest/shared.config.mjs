import { defineConfig } from 'vitest/config'

export default defineConfig({
    test: {
        environment: "jsdom",
        exclude: [
            "**/build/**",
            "**/dist/**",
            "**/node_modules/**",
            "**/@bundler/**",
            "**/@linter/**",
            "**/@test/**",
        ],
        include: [
            "**/*.unit.ts"
        ],
        globals: true,
    },
})
