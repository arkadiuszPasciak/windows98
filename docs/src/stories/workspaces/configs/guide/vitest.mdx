import { Meta, Title } from "@storybook/addon-docs/blocks";

<Meta title="Workspaces/Configs/Guide/Vitest" />

<Title>âš¡ Vitest</Title>

Complete guide to using Vitest configuration from `@windows98/configs` for unit and integration testing.

## Overview

Vitest configuration provides fast, modern testing setup optimized for TypeScript and modern JavaScript frameworks within the Windows98 ecosystem.

## Configuration Path

```
@windows98/configs/vitest/shared
```

### Install Dependencies

```bash
# Install vitest
pnpm add -D vitest

# Install the configs package
pnpm add -D @windows98/configs
```

## Basic Usage

### Simple Setup

```javascript
// ./@test/vitest/vitest.config.js
import { defineConfig } from 'vitest/config'
import sharedConfig from '@windows98/configs/vitest/shared'

export default defineConfig(sharedConfig)
```

### With Custom Settings

```javascript
// ./@test/vitest/vitest.config.js
import { defineConfig } from 'vitest/config'
import sharedConfig from '@windows98/configs/vitest/shared'

export default defineConfig({
  ...sharedConfig,
  test: {
    ...sharedConfig.test,
    environment: 'jsdom',
    setupFiles: ['./src/test-setup.ts']
  }
})
```

## Package.json Scripts

```json
{
  "scripts": {
	"test:unit:run": "vitest run -c ./@test/vitest/vitest.config.js",
	"test:unit:watch": "vitest -c ./@test/vitest/vitest.config.js",
  }
}
```

