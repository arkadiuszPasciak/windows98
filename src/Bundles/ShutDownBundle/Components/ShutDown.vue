<template>
  <UIModal
    class="ShutDown"
    :title="t('ShutDownBundle.title')"
    :resize-window="false"
    :width="330"
    :height="180"
    :modal-state="shutDownStore.modal"
    @close-modal="closeModal"
  >
    <div class="content">
      <img
        class="icon"
        width="32"
        height="32"
        src="src/Assets/Icons/Applications/mycomputer.png"
        alt="My computer"
      />
      <UIText class="description">{{ t('ShutDownBundle.description') }}</UIText>
      <UIRadio
        class="checkbox-shut-down"
        :name="shutDownRadioInputName"
        :checked="true"
        :model-value="EShutDownRadioCheck.SHUT_DOWN"
        @update:model-value="checkedRadioBoxes = $event"
      >
        {{ t('ShutDownBundle.shut-down') }}
      </UIRadio>
      <UIRadio
        class="checkbox-restart"
        :name="shutDownRadioInputName"
        :model-value="EShutDownRadioCheck.RESTART"
        @update:model-value="checkedRadioBoxes = $event"
      >
        {{ t('ShutDownBundle.restart') }}
      </UIRadio>
      <UIButton class="button is-ok" @click="triggerEvent">
        {{ t('ShutDownBundle.ok') }}
      </UIButton>
      <UIButton class="button is-cancel" @click="closeModal">
        {{ t('ShutDownBundle.cancel') }}
      </UIButton>
    </div>
  </UIModal>
</template>

<script setup lang="ts">
  import { ref, Ref } from 'vue'
  import { useI18n } from 'vue-i18n'
  import UIButton from '@Bundles/UIBundle/Components/UIButton.vue'
  import UIModal from '@Bundles/UIBundle/Components/UIModal.vue'
  import UIRadio from '@Bundles/UIBundle/Components/UIRadio.vue'
  import UIText from '@Bundles/UIBundle/Components/UIText.vue'
  import {
    restartSystem,
    shutDownSystem,
  } from '@Bundles/ShutDownBundle/Services/ShutDown.services'
  import { useShutDownStore } from '@Bundles/ShutDownBundle/Stores/ShutDown.stores'
  import {
    EShutDownRadioCheck,
    IShutDownRadioCheck,
    shutDownRadioInputName,
  } from '@Bundles/ShutDownBundle/Supports/ShutDown.supports'

  const { t } = useI18n()

  const shutDownStore = useShutDownStore()

  const checkedRadioBoxes = ref('shut-down') as Ref<IShutDownRadioCheck>

  const closeModal = (): void => {
    shutDownStore.updateModal(false)
  }

  const triggerEvent = () => {
    if (checkedRadioBoxes.value === EShutDownRadioCheck.RESTART) {
      restartSystem(window)
    } else if (checkedRadioBoxes.value === EShutDownRadioCheck.SHUT_DOWN) {
      shutDownSystem(window)
    } else {
      return
    }

    closeModal()
  }
</script>

<i18n src="@Bundles/ShutDownBundle/Locales/ShutDown.locales.json"></i18n>

<style lang="scss" scoped>
  @import '@Bundles/ShutDownBundle/Styles/ShutDown.styles.scss';
</style>
