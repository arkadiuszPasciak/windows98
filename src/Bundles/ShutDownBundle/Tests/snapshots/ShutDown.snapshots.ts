import { mount } from '@vue/test-utils'
import { expect, it, describe } from 'vitest'
import ShutDown from '@Bundles/ShutDownBundle/Components/ShutDown.vue'

const defaultWrapper = mount(ShutDown)

describe('[StartBundle]<Snapshots>(Start)', () => {
  it('renders correctly component before click button', () => {
    expect(defaultWrapper.element).toMatchSnapshot()
  })
})
