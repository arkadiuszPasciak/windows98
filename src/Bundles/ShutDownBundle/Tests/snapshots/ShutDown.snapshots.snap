// Vitest Snapshot v1

exports[`[StartBundle]<Snapshots>(Start) > renders correctly component before click button 1`] = `
<div
  class="UIModal move-window ShutDown"
  data-v-7ca490df=""
  data-v-99a75817=""
>
  <div
    class="container"
    data-v-99a75817=""
    style="width: 330px; height: 180px;"
  >
    <div
      class="header"
      data-v-99a75817=""
      style="cursor: default;"
    >
      <h3
        class="title"
        data-v-99a75817=""
      >
        Shut Down Windows
      </h3>
      <button
        class="UIButton size-small button-close"
        data-v-99a75817=""
        data-v-cf1bb3a9=""
        type="button"
      >
        
        <img
          alt="Close window"
          class="button-close-icon"
          data-v-99a75817=""
          height="7"
          src="/icons/close-modal.svg"
          width="8"
        />
        
        
        
        
        
      </button>
    </div>
    
    <div
      class="content"
      data-v-7ca490df=""
    >
      <img
        alt="My computer"
        class="icon"
        data-v-7ca490df=""
        height="32"
        src="/icons/applications/mycomputer.png"
        width="32"
      />
      <p
        class="UIText description"
        data-v-7ca490df=""
        data-v-f0b878d5=""
      >
        
        What do you want the computer to do?
        
      </p>
      <label
        class="UIRadio variant-primary checkbox-shut-down"
        data-v-7ca490df=""
        data-v-ccc35d9d=""
      >
        <input
          class="input"
          data-v-ccc35d9d=""
          name="shut-down-radio-inputs"
          type="radio"
          value="shut-down"
        />
        <span
          class="radio"
          data-v-ccc35d9d=""
        />
        <p
          class="UIText text"
          data-v-ccc35d9d=""
          data-v-f0b878d5=""
        >
          
          
          Shut Down
          
          
        </p>
      </label>
      <label
        class="UIRadio variant-primary checkbox-restart"
        data-v-7ca490df=""
        data-v-ccc35d9d=""
      >
        <input
          class="input"
          data-v-ccc35d9d=""
          name="shut-down-radio-inputs"
          type="radio"
          value="restart"
        />
        <span
          class="radio"
          data-v-ccc35d9d=""
        />
        <p
          class="UIText text"
          data-v-ccc35d9d=""
          data-v-f0b878d5=""
        >
          
          
          Restart
          
          
        </p>
      </label>
      <button
        class="UIButton size-medium button is-ok"
        data-v-7ca490df=""
        data-v-cf1bb3a9=""
        type="button"
      >
        
        
        
        OK
        
        
        
      </button>
      <button
        class="UIButton size-medium button is-cancel"
        data-v-7ca490df=""
        data-v-cf1bb3a9=""
        type="button"
      >
        
        
        
        Cancel
        
        
        
      </button>
    </div>
    
  </div>
</div>
`;
